### PR TITLE
Added requirement for 4B accesses to be atomic if 4B aligned

### LINKED ISSUE #34: Clarify memory access behavior
from @ved-rivos:

This non-normative text "Therefore, implementations are not required to ensure atomicity when storing to memory. However, writing to both registers written by a 64-bit load must happen atomically to ensure fault handling is possible." implies that the 64-bit load needs to be atomic. This could be reworded to "For the purpose of raising misaligned-address or access exceptions, the address is considered to be naturally aligned if it is a multiple of 8. Even if naturally aligned, the memory access might not be performed atomically. If the address is a multiple of 4, then each 4-byte access is performed atomically. The LD instruction must however write the loaded data to the pair of destination registers atomically to ensure fault handling is possible."

### PR DESCRIPTION
**Pull request by christian-herber-nxp**

### CODE REVIEW

File: zilsd.adoc [hunk 1]
```diff
@@ -35,12 +35,17 @@ The Zilsd extension adds the following RV32-only instructions:
 
 [NOTE]
 ====
-As the access size is 64-bit, accesses are only considered naturally aligned for effective addresses that are a multiple of 8. In this case, these instruction are guaranteed to not raise an address-misaligned exception.
+As the access size is 64-bit, accesses are only considered naturally aligned for effective addresses that are a multiple of 8.
+In this case, these instruction are guaranteed to not raise an address-misaligned exception.
```
Comment (ved-rivos) [inline zilsd.adoc:39]: or an access fault exception.

**Review by ved-rivos — APPROVED** [review]

File: zilsd.adoc [hunk 2]
```diff
@@ -35,12 +35,17 @@ The Zilsd extension adds the following RV32-only instructions:
 
 [NOTE]
 ====
-As the access size is 64-bit, accesses are only considered naturally aligned for effective addresses that are a multiple of 8. In this case, these instruction are guaranteed to not raise an address-misaligned exception.
+As the access size is 64-bit, accesses are only considered naturally aligned for effective addresses that are a multiple of 8.
```
Comment (ved-rivos) [inline zilsd.adoc:38]: Should also add:
For the purpose of raising misaligned-address or access exceptions, the address is considered to be naturally aligned if it is a multiple of 8.

**Comment by ved-rivos** [comment]
LGTM

File: zilsd.adoc [hunk 3]
```diff
@@ -35,12 +35,17 @@ The Zilsd extension adds the following RV32-only instructions:
...
+In this case, these instruction are guaranteed to not raise an address-misaligned exception.
+Even if naturally aligned, the memory access might not be performed atomically.
```
Comment (ved-rivos) [inline zilsd.adoc:38]: This text should be normative.

**Review by tovine — APPROVED** [review]

File: zilsd.adoc [hunk 4]
```diff
@@ -98,19 +100,31 @@ It is implementation defined whether interrupts can also be taken during the seq
...
 * store instructions:
-** A sequence of one or more stores writing the bytes of the double word
-*** The bytes may be stored in any order.
-*** The bytes may be grouped into larger accesses.
-*** Any of the bytes may be stored multiple times.
+** A sequence of one or more stores writing the bytes of the doubleword
+*** If the effective address is 4B aligned:
+**** The bytes are grouped into word accesses.
+**** The words may be stored in any order.
+**** The words may be grouped into larger accesses.
```
Comment (ved-rivos) [inline zilsd.adoc]: Lets tighten this a bit. The words may be grouped into a doubleword. Just to avoid someone creating a 5B + 3B grouping.

Comment (christian-herber-nxp) [reply zilsd.adoc]: that was unintended copy paste error. fixed

Comment (ved-rivos) [reply zilsd.adoc]: should also be fixed for the stores.

Comment (christian-herber-nxp) [reply zilsd.adoc]: sorry, i did fix it but failed to push. i think it looks fine now

**Pull request merged**